.github/workflows/builder_docker-based_slsa3.yml

```diff
@@ -0,0 +1,191 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: SLSA Docker-based builder
+
+# TODO: permissions
+
+# TODO: env
+
+defaults:
+  run:
+    shell: bash
+
+###################################################################
+#                                                                 #
+#            Input and output argument definitions                #
+#                                                                 #
+###################################################################
+on:
+  workflow_call:
+    inputs:
+      builder-image:
+        description: >
+          The OCI image name of the builder image responsible for 
+          creating the artifact. This must not include a tag or digest.
+        required: true 
+        type: string
+      builder-digest:
+        description: >
+          The OCI image digest of the builder-image. 
+          This must be specified in order to fully specify the builder 
+          image. The image digest is of the form '<alg>:<digest>'
+        required: true 
+        type: string
+      config:
+        description: >
+          Path to a configuration file containing a command that the builder
+          image should be invoked with and optionally, docker options
+          and environment variables.
+        required: true 
+        type: string
+      output:
+        description: >
+          Path to the output artifact(s) relative to the root of the Git
+          repository.
+          Users may specify a single artifact or use a shell-style wildcard
+          pattern to match multiple artifacts.
+        required: true 
+        type: string
+      source-repository:
+        description: >
+          Source repository to checkout for the build. For example, octocat/hello-world.
+          The builder will mount the root of this source repository to the workspace.
+
+          By default, uses the source repository of the caller workflow.
+        required: false 
+        type: string
+      source-digest:
+        description: >
+          The SHA of the source repository to checkout. 
+
+          By default, uses the source repository of the caller workflow.
+        required: false
+        type: string
+      # TODO: Options for requesting and verifying provenance of builder image
+      # This will use slsa-verifier and slsa-verifier option.
+      #      * request-builder-verification
+
+      # TODO: Authentication options for the builder-image.
```

Comment:
Ideally the image should be publicly available, no?

Reply:
not if it's used by a private repository. We've had several users reach out to us who used private repos.

Reply:
Yes. I agree. I hope it is possible to specify this kind of dependency between input option on GitHub workflows.

Reply:
I added TODOs here to specify secrets and also a TODO step in the build to authenticate first if needed. For now we can start with public options as we test